### ISSUE TITLE
OpenGL safety

### ISSUE THREAD

**Issue by davidpiuva** [magic-lang/ooc-kean#1427]
* Prevent any input texture from also being the output texture. This may crash on phones.
* Prevent any shader from reading a texture that has not been assigned recently.
* Prevent uniforms from leaking to another draw call.

**Comment by davidpiuva** [comment]
@emilwestergren @sebastianbaginski @fredrikbryntesson any suggestions?

**Comment by emilwestergren** [comment]
I don't think this should be the priority right now, but we can keep the issue and discuss it later.
